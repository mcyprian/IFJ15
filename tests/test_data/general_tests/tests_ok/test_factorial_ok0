int factorial(int n)
{
    int tmp;
    auto decrement = n - 1;

    if(n  < 2)
    {
        return 1;
    }
    else
    {
        tmp = factorial(decrement);
        return n*tmp;
    }
}

int main()
{
    int a; int res;

    a = 6;
    res = factorial(a);

    auto neg =  0 - res;

    cout << "faktorial: " << res << "\n";
    cout << "negovany: " << neg << "\n";

    return 42;
}
