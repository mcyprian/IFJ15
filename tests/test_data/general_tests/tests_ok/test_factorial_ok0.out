faktorial: 720
negovany: -720
